tests/phpunit/CRM/MembershipExtras/Service/MembershipInstalmentsScheduleTest.php

```diff
@@ -644,6 +644,30 @@ public function testExceptionIsThrownIfMembershipTypeDurationUnitIsDay() {
     $this->getMembershipInstalmentsSchedule([$membershipType], MembershipInstalmentsSchedule::ANNUAL);
   }
 
+  /**
+   * Tests that instalments date are calculated properly and months follows a sequential order.
+   */
+  public function testRollingInstalmentDatesAreCalculatedProperly() {
+    $currentMonth = 1;
+    $startDate = new DateTime('2022-01-31');
+    $membershipTypes = $this->mockRollingMembershipTypes();
+    $schedule = $this->getMembershipSchedule(
+      $membershipTypes,
+      MembershipInstalmentsSchedule::MONTHLY,
+      $startDate
+    );
+
+    foreach ($schedule['instalments'] as $index => $instalment) {
+      $instalmentDate = $instalment->getInstalmentDate();
+      $year = $instalmentDate->format('Y');
+      $month = $instalmentDate->format('n');
+      $day = $instalmentDate->format('d');
+
+      $this->assertTrue(checkdate($month, $day, $year));
+      $this->assertEquals($month, $currentMonth++);
```

Comment:
testing the month is correct is not enough to make sure the date is calculated correctly, a more accurate test should include an array that include all the expected installment dates (all hardcoded), and you compare the resulted dates against them.

Reply:
Thanks, @omarabuhussein.

> a more accurate test should include an array that include all the expected installment dates (all hardcoded)

I avoided doing this, since we have tests like `testRollingMonthlyInstalmentAmounts` and others in place, and  I thought they were enough to assert the correctness of the returned date.

e.g. if the `start_date` is 2022-01-31 and the next date is 2022-03-01 the test will fail.